src/mngt/ocf_mngt_cache.c

```diff
@@ -2052,60 +2043,70 @@ static void _ocf_mngt_standby_init_properties(ocf_pipeline_t pipeline,
 	ocf_pipeline_next(pipeline);
 }
 
-static void _ocf_mngt_load_metadata_unsafe(ocf_pipeline_t pipeline,
+static void _ocf_mngt_standby_preapre_mempool(ocf_pipeline_t pipeline,
 		void *priv, ocf_pipeline_arg_t arg)
 {
 	struct ocf_cache_attach_context *context = priv;
 	ocf_cache_t cache = context->cache;
+	int result;
 
-	ocf_cache_log(cache, log_info, "Loading cache state...\n");
-	ocf_metadata_load_unsafe(cache,
-			_ocf_mngt_load_unsafe_complete, context);
+	result = ocf_metadata_passive_io_ctx_init(cache);
+	if(!result)
+		context->flags.pio_mpool = true;
+
+	OCF_PL_NEXT_ON_SUCCESS_RET(context->pipeline, result);
 }
 
-static void _ocf_mngt_bind_preapre_mempool(ocf_pipeline_t pipeline,
+static void _ocf_mngt_standby_init_structures_attach(ocf_pipeline_t pipeline,
 		void *priv, ocf_pipeline_arg_t arg)
 {
 	struct ocf_cache_attach_context *context = priv;
 	ocf_cache_t cache = context->cache;
-	int result;
 
-	result = ocf_metadata_passive_io_ctx_init(cache);
-	if(!result)
-		context->flags.pio_mpool = true;
+	init_attached_data_structures_recovery(cache, true);
+	__populate_free_safe(cache);
 
-	OCF_PL_NEXT_ON_SUCCESS_RET(context->pipeline, result);
+	ocf_pipeline_next(pipeline);
 }
 
-static void _ocf_mngt_bind_init_attached_structures(ocf_pipeline_t pipeline,
+static void _ocf_mngt_standby_init_structures_load(ocf_pipeline_t pipeline,
 		void *priv, ocf_pipeline_arg_t arg)
 {
 	struct ocf_cache_attach_context *context = priv;
 	ocf_cache_t cache = context->cache;
-	int result;
 
 	init_attached_data_structures_recovery(cache, false);
 
+	ocf_pipeline_next(pipeline);
+}
+
+static void _ocf_mngt_standby_init_pio_concurrency(ocf_pipeline_t pipeline,
+		void *priv, ocf_pipeline_arg_t arg)
+{
+	struct ocf_cache_attach_context *context = priv;
+	ocf_cache_t cache = context->cache;
+	int result;
+
 	result = ocf_pio_concurrency_init(&cache->standby.concurrency, cache);
 	if (!result)
 		context->flags.pio_concurrency = true;
 
 	OCF_PL_NEXT_ON_SUCCESS_RET(context->pipeline, result);
 }
 
-static void _ocf_mngt_bind_recovery_unsafe(ocf_pipeline_t pipeline,
+static void _ocf_mngt_standby_recovery_unsafe(ocf_pipeline_t pipeline,
```

Comment:
I think all the "unsafe" variants of recovery functions in this file are unnecessary - we do not need to handle invalid status bits combinations or stale mappings

Reply:
I mean __populate_free_unsafe and _ocf_mngt_rebuild_metadata(initialized = false)

Reply:
Done.